frame/worklet.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Utilities for running code in a worklet-like environment.
+ * (Browser-native implementations of FLEDGE will use real worklets, but
+ * JavaScript code can't create custom worklets, so we use Web Workers instead.)
+ */
+
+import { isArray } from "../lib/shared/guards";
+import { awaitMessageToWorker } from "../lib/shared/messaging";
+import { BidData, CanonicalInterestGroup } from "./types";
+
+/**
+ * Interprets `biddingScript` as JavaScript code and runs it in a worklet-like
+ * Web Worker environment, then calls its `generateBid` function and returns the
+ * result, or null if any exceptions or validation errors occur. (Such errors
+ * are logged, either inside or outside the Web Worker.)
+ *
+ * @param group The originating interest group, passed through to generateBid.
```

Comment:
Can you add some comments about how this is not intended to be used with untrusted JS and does not provide protection against exfiltration?